src/main/java/org/jenkinsci/plugins/edithelp/EditHelpRootAction.java

```diff
@@ -0,0 +1,155 @@
+package org.jenkinsci.plugins.edithelp;
+
+import java.io.*;
+
+import hudson.Extension;
+import hudson.model.Action;
+import hudson.model.RootAction;
+import hudson.model.View;
+import jenkins.model.Jenkins;
+import org.apache.tools.ant.taskdefs.Recorder;
+import org.kohsuke.stapler.Stapler;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.lang.Klass;
+import sun.misc.URLClassPath;
+
+import javax.swing.*;
+import java.net.Socket;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.TreeMap;
+
+
+@Extension
+public class EditHelpRootAction implements RootAction {
+
+    //reading the catalog of helpmanager
+    private static ArrayList<File> listWithFileNames = new ArrayList<>();
+    public static void getListFiles(String str) {
```

Comment:
🐜 Unsynchronized method in public API. In your case it may blow UP if Jenkins constructs two instances of your action if it gets extended in another plugin (corner case)